### PR TITLE
ARXIVNG-1375 email confirmation upon finalization

### PR DESCRIPTION
**Pull request by erickpeirson**
This implements sending a confirmation email when a submission is finalized. 

Unrelated: I also tweaked the way that async tasks are run, so that we can follow up on monitoring tasks (e.g. during compilation) more reliably. More on that when I implement in the UI.

To see this in action, head over to https://github.com/arXiv/arxiv-submission-ui/pull/62.

This also depends on https://github.com/arXiv/arxiv-base/pull/108

### CODE REVIEW

**Comment by erickpeirson** [comment]
This is really a UI concern; closing for now

**Comment by erickpeirson** [comment]
Here is the meat of the PR: https://github.com/arXiv/arxiv-submission-core/pull/58/files#diff-d3fd5a28718d48fcc0dd9cc16b12ce93R19

File: core/arxiv/submission/rules/email_notifications.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        mail.send(event.creator.email, "Submission to arXiv received",
+                  render_template("submission-core/confirmation-email.txt",
+                                  **context),
+                  render_template("submission-core/confirmation-email.html",
+                                  **context))
+    return []
+
+
+def email_is_enabled() -> bool:
+    """Whether or not a Jaccard similarity is above the threshold."""
```
Comment (eawoods) [inline core/arxiv/submission/rules/email_notifications.py]: why do we need to know whether a Jaccard similarity is above the threshold (application to whether or not email is enabled)? What is being compared?

File: core/arxiv/submission/templates/submission-core/confirmation-email.html [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  comments = submission.metadata.comments,
+  msc_class = submission.metadata.msc_class,
+  acm_class = submission.metadata.acm_class,
+  journal_ref = submission.metadata.journal_ref,
+  doi = submission.metadata.doi,
+  report_num = submission.metadata.report_num,
+  version = submission.version,
+  submission_history = [],
+  secondary_categories = submission.secondary_categories,
+  embed_stylesheet=1) }}
```
Comment (eawoods) [inline core/arxiv/submission/templates/submission-core/confirmation-email.html]: as mentioned in submission-ui, it is safer to include this information as an unstyled section or possibly to write an inliner for styles. It's great info and more than we have now, worth including and possibly better to reduce delivery or display errors by doing plaintext data on this for all email. (with maybe a lightweight graphic element to set it off from the main body?)

Comment (eawoods) [reply core/arxiv/submission/templates/submission-core/confirmation-email.html]: Also should label this as data that the user submitted. And test against TeX, or possibly might need to remove some of it for logistical simplicity.

File: core/arxiv/submission/rules/email_notifications.py (hunk 1 above)
Comment (erickpeirson) [reply core/arxiv/submission/rules/email_notifications.py]: Oh yuck, how sloppy

File: core/arxiv/submission/templates/submission-core/confirmation-email.html (hunk 2 above)
Comment (erickpeirson) [reply core/arxiv/submission/templates/submission-core/confirmation-email.html]: Ah crap, didn't think about TeX; maybe we can pre-render? Let's look at this

**Comment by erickpeirson** [comment]
Per discussion, removed abs preview. Will add link to preview as part of [ARXIVNG-2006](https://arxiv-org.atlassian.net/browse/ARXIVNG-2006)

**Pull request merged**